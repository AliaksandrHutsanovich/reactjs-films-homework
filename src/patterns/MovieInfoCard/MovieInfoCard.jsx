import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Card, Layout } from 'antd';
import { CloseOutlined } from '@ant-design/icons';
import renderTitle from './utils/renderTitle';

import '!style-loader!css-loader!sass-loader!antd/dist/antd.css'; //eslint-disable-line

import MovieDescription from '../MovieDescription';
import { Text, Button } from '../../components';

import styles from './MovieInfoCard.scss';

const { Meta } = Card;

const MovieInfoCard = ({
  id,
  title,
  genres,
  rating,
  info,
  onClose,
  onDialogOpen,
  onScrollInfo,
}) => (
  <Card className={styles.card}>
    <Meta
      description={(
        <Layout className={styles.card__content}>
          <CloseOutlined
            onClick={onClose}
            className={styles.icon}
          />
          <MovieDescription
            id={id}
            title={title}
            genres={genres}
            rating={rating}
            ratingFontType="bordered"
            titleFontType="white"
            renderTitle={renderTitle}
          />
          <Layout
            className={styles.card__textWrapper}
            onScroll={onScrollInfo}
          >
            <Text type="white" size={12}>
              {info}
            </Text>
          </Layout>
          <Button
            title="Watch Now"
            onClick={onDialogOpen}
          />
        </Layout>
      )}
    />
  </Card>
);

MovieInfoCard.propTypes = {
  title: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(
    PropTypes.string,
  ).isRequired,
  rating: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  info: PropTypes.string,
  onClose: PropTypes.func.isRequired,
  onDialogOpen: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  onScrollInfo: PropTypes.func.isRequired,
};

MovieInfoCard.defaultProps = {
  info: 'No description about movie',
};

export default memo(MovieInfoCard);
