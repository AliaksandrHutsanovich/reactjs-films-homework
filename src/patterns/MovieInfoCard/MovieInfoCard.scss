@import '../../styles/global';

.card {
  width: #{$cardWidth}px;
  height: #{$cardHeight}px;
  background-color: rgba(0, 0, 0, 0.5) !important;
  border-radius: 4px 4px 0 0;

  &__content {
    background-color: transparent !important;
    &>:nth-child(n + 2) {
      margin: 12px 0;
    }
  }

  &__textWrapper {
    background-color: transparent !important;
    max-height: 105px;
    overflow: auto;
  }

  button {
    width: fit-content;
  }
}

.icon {
  width: fit-content;
  color: white;
  transition: color 1s ease-in;
  transition:  background-color 1s linear;
  &:hover {
    color: yellow;
    background-color: red;
  }
}
