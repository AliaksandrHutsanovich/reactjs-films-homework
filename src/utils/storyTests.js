const storyTests = (stories, renderer) => {
  Object.keys(stories).forEach((storyKey) => {
    if (typeof stories[storyKey] !== 'function') {
      return;
    }
    it(`${storyKey} should be rendered properly`, () => {
      renderer.render(stories[storyKey]());
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
};

export default storyTests;
