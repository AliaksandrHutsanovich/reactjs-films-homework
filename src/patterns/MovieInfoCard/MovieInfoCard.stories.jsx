import React from 'react';
import { action } from '@storybook/addon-actions';
import MovieInfoCard from './MovieInfoCard';

export default {
  component: MovieInfoCard,
  title: 'patterns/MovieInfoCard',
  decorators: [
    (story) => (
      <div
        style={{
          backgroundColor: 'rgba(0, 0, 0, 0.5)',
          width: 'fit-content',
          borderRadius: '6px 6px 0 0',
        }}
      >
        {story()}
      </div>
    ),
  ],
};

const genres = [
  'Action',
  'Adventure',
  'Fantasy',
];

const info = `
  There are growing dangers in the
  wizarding world of 1926 New York.
  Something misterious is leaving a
  path of destructions in the streets,
  threatening to expose the wizarding.
`;

export const cardExample = () => (
  <MovieInfoCard
    title="ASSASIN'S CREED"
    genres={genres}
    rating="4.7"
    info={info}
    onClose={action('the close icon was clicked')}
    onDialogOpen={action('the button for opening dialog was clicked')}
  />
);
