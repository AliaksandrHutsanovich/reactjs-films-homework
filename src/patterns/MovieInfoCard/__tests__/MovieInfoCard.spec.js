import ShallowRenderer from 'react-test-renderer/shallow';
import storyTests from '../../../utils/storyTests';
import * as stories from '../MovieInfoCard.stories';

describe('<MovieInfoCard />', () => {
  const renderer = new ShallowRenderer();

  storyTests(stories, renderer);
});
