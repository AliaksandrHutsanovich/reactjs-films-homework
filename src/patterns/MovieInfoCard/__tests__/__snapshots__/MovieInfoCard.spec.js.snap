// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieInfoCard /> cardExample should be rendered properly 1`] = `
<Card
  className="card"
>
  <Meta
    description={
      <Layout
        className="card__content"
      >
        <ForwardRef(CloseOutlined)
          className="icon"
          onClick={[Function]}
        />
        <UNDEFINED
          genres={
            Array [
              "Action",
              "Adventure",
              "Fantasy",
            ]
          }
          id={1}
          rating="4.7"
          ratingFontType="bordered"
          renderTitle={[Function]}
          title="ASSASIN'S CREED"
          titleFontType="white"
        />
        <Layout
          className="card__textWrapper"
        >
          <UNDEFINED
            size={12}
            type="white"
          >
            
  There are growing dangers in the
  wizarding world of 1926 New York.
  Something misterious is leaving a
  path of destructions in the streets,
  threatening to expose the wizarding.

          </UNDEFINED>
        </Layout>
        <UNDEFINED
          onClick={[Function]}
          title="Watch Now"
        />
      </Layout>
    }
  />
</Card>
`;
